packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -0,0 +1,350 @@
+    frequentistVariance(
+      v,
+      mean,
+      n / nVariations,
+      v,
+      mean * (1.0 + effectSize),
+      n / nVariations,
+      true,
+    ),
+  );
+}
+
+/**
+ * Performs power calculation
+ *
+ * @param effectSize Scalar lift (relative to the scalar mean of the distribution, expressed as percentage).
+ * @param mean Scalar mean of the distribution.
+ * @param variance Scalar variance of the distribution.
+ * @param n Scalar sample size.
+ * @param nVariations Scalar number of variations.
+ * @param alpha false positive rate (default: 0.05).
+ * @param twoTailed Binary indicator if the test is 1 or 2-tailed (default: true).
+ * @returns Estimated power.
+ */
+export function powerEst(
+  effectSize: number,
+  mean: number,
+  variance: number,
+  n: number,
+  nVariations: number,
+  alpha: number = 0.05,
+  twoTailed: boolean = true,
+  sequentialTuningParameter = 0,
+): number {
+  if (typeof twoTailed !== "boolean") {
+    throw new Error("twoTailed must be boolean.");
+  }
+  const zStar = twoTailed
+    ? normal.quantile(1.0 - 0.5 * alpha, 0, 1)
+    : normal.quantile(1.0 - alpha, 0, 1);
+
+  let standardError = 0;
+  if (sequentialTuningParameter > 0) {
+    standardError = sequentialPowerStandardError(
+      variance,
+      mean,
+      n,
+      nVariations,
+      effectSize,
+      alpha,
+      sequentialTuningParameter,
+    );
+  } else {
+    standardError = powerStandardError(
+      variance,
+      mean,
+      n,
+      nVariations,
+      effectSize,
+    );
+  }
+  const standardizedEffectSize = effectSize / standardError;
+  const upperCutpoint = zStar - standardizedEffectSize;
+  let power = 1 - normal.cdf(upperCutpoint, 0, 1);
+  if (twoTailed) {
+    const lowerCutpoint = -zStar - standardizedEffectSize;
+    power += normal.cdf(lowerCutpoint, 0, 1);
+  }
+  return power;
+}
+
+/**
+ * Calculates minimum detectable effect
+ *
+ * @param power desired power.
+ * @param mean Scalar mean of the distribution.
+ * @param variance Scalar variance of the distribution.
+ * @param n Scalar sample size.
+ * @param nVariations Scalar number of variations.
+ * @param alpha false positive rate (default: 0.05).
+ * @returns Estimated power.
+ */
+export function findMde(
+  power: number,
+  mean: number,
+  variance: number,
+  n: number,
+  nVariations: number,
+  alpha: number = 0.05,
+  sequentialTuningParameter = 0,
+): MDEResults {
+  // Error handling:
+  if (power <= alpha) {
+    const mdeResults: MDEResults = {
+      type: "error",
+      description: "power must be greater than alpha.",
+    };
+    return mdeResults;
+  }
+  const nA = n / nVariations;
+  const z =
+    normal.quantile(1.0 - 0.5 * alpha, 0, 1) -
+    normal.quantile(1.0 - power, 0, 1);
+  let v = variance;
+  if (sequentialTuningParameter > 0) {
+    v = sequentialPowerSequentialVariance(
+      variance,
+      nA,
+      alpha,
+      sequentialTuningParameter,
+    );
+  }
+  if (nA <= (v * z ** 2) / (2 * mean ** 2)) {
```

Comment:
Maybe add a comment about what this check is actually looking for? It's a little opaque to me.

Reply:
done